eloquentjs/11_1_quiet_times/quiet_times.js

```diff
@@ -0,0 +1,47 @@
+const { generateLogs } = require('./src/generateLogs');
+const { textFile } = require('./src/textFile');
+
+/**
+ * Returns promise of hourly security camera activity recorded for given day of week (0-6)
+ * @param {*} day 
+ * @returns 
+ */
+async function activityTable(day) {
+    const camera_logs = await textFile('camera_logs.txt');
+    const filePaths = camera_logs.split('\n').filter(j => j);
+
+    const timeStamps = [];
+    for (let i = 0; i < filePaths.length; i++) {
+        const logContents = await textFile(`weekly_logs/${filePaths[i]}`);
+
+        logContents.split('\n').filter(j => j).forEach(element => {
+            timeStamps.push(element);
+        });
+    }
+
+    const activity = Array(24).fill(0);
+
+    timeStamps.forEach((time) => {
+        const date = new Date();
+        date.setTime(time);
+
+        if (date.getDay() === day) {
+            const hour = date.getHours();
+            activity[hour]++;
+        }
+    });
+
+    return new Promise((resolve) => {
```

Comment:
You should be able to just return activity. Because this is an async function it will get wrapped into a promise for you by the runtime.